### DISCUSSION TITLE
gptneox.cpp for OpenAssistant models

### DISCUSSION THREAD

**Discussion by byroneverson** [ggml-org/llama.cpp#1202] [Show and tell]
Put together a quick fork of llama.cpp for using OpenAssistant StableLM and Pythia models. So far the 7B stablelm and 12B pythia models are supported but I'm sure I will add more soon as they are made available. There are some easy to use scripts in the scripts directory for those of you that want to check it out with ease. Refer to the README-GPTNEOX for more details.

https://github.com/byroneverson/gptneox.cpp

**Comment by Green-Sky** [comment]
cool, gg's implementation seems to have an error. would be nice if you could use your experience with neox and apply it https://github.com/ggerganov/ggml/tree/master/examples/stablelm#warning

**Reply by Green-Sky** [reply]
see: https://github.com/ggerganov/llama.cpp/issues/1063#issuecomment-1525975657

**Reply by byroneverson** [reply]
Looks like a fun challenge, I will dig into it once I am home if it is not already resolved by then.
